### PR TITLE
[Doc] reorganize show tables and optimize others

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-definition/CREATE EXTERNAL CATALOG.md [hunk 1]
```diff
@@ -24,7 +27,7 @@ PROPERTIES ("key"="value", ...)
 
 | **Parameter** | **Required** | **Description**                                              |
 | ------------- | ------------ | ------------------------------------------------------------ |
-| catalog_name  | Yes          | The name of the external catalog. The naming conventions are as follows:<ul><li>The name can contain letters, digits (0-9), and underscores (_). It must start with a letter.</li><li>The name cannot exceed 64 characters in length.</li></ul> |
+| catalog_name  | Yes          | The name of the external catalog. The naming conventions are as follows:<ul><li>The name can contain letters, digits (0-9), and underscores (_). It must start with a letter.</li><li>The name is case-sensitive and cannot exceed 64 characters in length.</li></ul> |
```
Comment (EsoragotoSpirit) [inline docs/sql-reference/sql-statements/data-definition/CREATE EXTERNAL CATALOG.md:30]: 1023？

File: docs/sql-reference/sql-statements/data-manipulation/SHOW TABLES.md [hunk 2]
```diff
@@ -2,10 +2,81 @@
...
+| emps2                      |
++----------------------------+
+```
+
+Example 2: View tables in database `hudi_db` of the external catalog `hudi_catalog`.
+
+```plain
+show tables from hudi_catalog.hudi_db;
++----------------------------+
+| Tables_in_hudi_db       |
```
Comment (EsoragotoSpirit) [inline docs/sql-reference/sql-statements/data-manipulation/SHOW TABLES.md:69]: 少了空格

File: docs/sql-reference/sql-statements/data-manipulation/SHOW CATALOGS.md [hunk 3]
```diff
@@ -18,27 +18,30 @@ SHOW CATALOGS
...
-The following table describes the parameters returned by this statement.
+The following table describes the fields returned by this statement.
 
-| **Parameter** | **Description**                                              |
+| **Field** | **Description**                                              |
 | ------------- | ------------------------------------------------------------ |
 | Catalog       | The catalog name.                                            |
-| Type          | The catalog type.                                            |
-| Comment       | The comments of a catalog.StarRocks does not support adding comments to an external catalog. Therefore, the parameter value is `NULL` for an external catalog. If the returned catalog is `default_catalog`, the parameter value is `Internal Catalog`. `default_catalog` is the only internal catalog in a StarRocks cluster. |
+| Type          | The catalog type. `Internal` is returned if the catalog is `default_catalog`. The corresponding catalog name is returned if the catalog is an external catalog, such as `Hive`, `Hudi`, or `Iceberg`. |
```
Comment (EsoragotoSpirit) [inline docs/sql-reference/sql-statements/data-manipulation/SHOW CATALOGS.md:26]: The corresponding catalog type？

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by evelynzhaojie** [comment]
@mergify backport branch-3.1 branch-3.0

**Pull request merged**